src/exception/tools_exception.py

```diff
@@ -0,0 +1,25 @@
+class ToolsException:
```

Comment:
I don't think these custom Exceptions are being used the right way. With a custom exception:

```python
def CustomException(Exception):
    def __init__(self, message: str):
        pass
```

You would raise this exception when there are issues that meet that Exception. Exceptions like `TypeError` is a good example.

```python
# raise TypeError if object is not correct type
if not isinstance(obj, str):
    raise TypeError("Object must be str.")
```

The Exceptions you are defining are not able to be caught and excepted for, which defeats the purpose of an exception because you are doing `exit()` with a status code and message.

Reply:
Oh totally, was my plan after adding everything that is currently there from my branch bfreeze-cleanup. Doing it in small chunks at the time being and including things that are imported alongside the specific parts I am doing pull requests for my own sanity. Once when everything is converted over and the previous build can be broken off in the qcpy dir, then the actual implementation can be done of redoing the exceptions. If you just want me to remove this chunk until it is fully flushed out then that works with me as well.

Reply:
Oh okay, that makes sense. I can go in and clean that up after you are done migrating the work you've been doing.

Reply:
Sure that works with me! Thank you for doing that.